plugins/mongodb/examples/mongodb-test.ecl

```diff
@@ -0,0 +1,188 @@
+        ]
+    );
+ENDEMBED;
+
+dataset({STRING _id, Integer count}) countPropertyTypes(INTEGER cntOrd, INTEGER idOrd) := Embed(mongodb : user(user), password(pwd), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    aggregate(
+        [
+            { $group: { _id: "$property_type", count: { $count: {} } } },
+            { $sort: { count: $cntOrd, _id: $idOrd } }
+        ]
+    );
+ENDEMBED;
+
+dataset(layouttransactions) findTransactions() := Embed(mongodb : user(user), password(pwd), server(server), database('sample_analytics'),  collection('transactions'))
+    find({});
+ENDEMBED;
+
+dataset(layoutDates) findTransactionDates() := Embed(mongodb : user(user), password(pwd), server(server), database('sample_analytics'),  collection('transactions'))
+    find({});
+ENDEMBED;
+
+dataset(layoutFees) findAndSort(REAL4 max, REAL4 min, INTEGER asc) := Embed(mongodb : user(user), password(pwd), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    aggregate([{$match: { price: { $gte: $min, $lt: $max}}}, {$sort: {price: $asc, extra_people: $asc, security_deposit: $asc}}]);
+ENDEMBED; 
+
+// dataset({INTEGER inserted_count}) insertMany(dataset(layoutEmployee) employees) := Embed(mongodb : user(user), password(pwd), server(server), database('mydb'),  collection('test2'))
+//     insert({$employees});
+// ENDEMBED;
+// employeeDS := DATASET ([{1, 'John', 'Andrews', 101000.5}, {2, 'Anne', 'Smith', 100000.7}, {3, 'Amy', 'Isaac', 103000.1}, {4, 'Kirk', 'Captain', 109000.9}, {5, 'Steve', 'Rogers', 99000.6}, 
+//             {6, 'Evan', 'Bosch', 104000.5}, {7, 'Jack', 'Adams', 101000.5}, {8, 'Vince', 'Carter', 306000.5}, {9, 'Beth', 'Stevens', 102000.2}, {10, 'Samantha', 'Rogers', 107000.5}], layoutEmployee);
+
+// boolean createIndex(INTEGER asc) := Embed(mongodb : user(user), password(pwd), server(server), database('mydb'),  collection('test2'))
+//     create_index({ "first": $asc, "last": $asc});
+// ENDEMBED;
+
+SEQUENTIAL
+(
+    OUTPUT(getCount(), NAMED('CountAllDocuments'));
+    // OUTPUT(EmployeeAggregate(), NAMED('EmployeeAggregate'));
+    OUTPUT(findABInfo(beds, accommodates), NAMED('AirbnbInformation'));
+    OUTPUT(findCount(nights), NAMED('CountAggregate'));
+    OUTPUT(findBedType(), NAMED('CountRealBeds'));
+    OUTPUT(searchPropertySorted(houseType, 1), NAMED('SearchPropertiesAsc'));
+    OUTPUT(searchPropertySorted(houseType, -1), NAMED('SearchPropertiesAndDec'));
+    OUTPUT(countPropertyTypes(-1, 1), NAMED('CountPropertyTypes'));
+    OUTPUT(CHOOSEN(findTransactions(), 5), NAMED('FindAllTransactions'));
+    OUTPUT(findTransactionDates(), NAMED('TransactionPeriods'));
+    OUTPUT(findAndSort(80.75, 20.50, -1), NAMED('FindAndSortByPrice'));
+    // OUTPUT(insertMany(employeeDS), NAMED('InsertMany'));
+    // OUTPUT(createIndex(1), NAMED('CreateIndexOnNames'));
+    OUTPUT('Done', Named('Status'));
+);
+
+// $or is not allowed in the M0 tier of MongoDB atlas
```

Comment:
Commented-out code is useful or no?

Reply:
I thought it would be useful to have more examples, but for functions that made changes to the database I had those commented out just so I wasn't constantly updating the database. The $or function I thought would be good to show how complex it can get, but I can't run it on my tier so I commented it out.